main.py

```diff
@@ -4,57 +4,110 @@
 
 import google.generativeai as genai
 
+from src.pipeline.batch_inference import gen_local_lm_responses
 from src.pipeline.eval import eval_on_records
-from utils import validate_steps
+from src.pipeline.utils import push_to_hf_hub
+
+def batch_inference(args):
+    hf_hub = False
 
-async def main(args):
-    if args.gemini_api_key is not None:
-        genai.configure(api_key=args.gemini_api_key)
-    
     if args.load_in_8bit is True \
         and args.load_in_4bit is True:
         raise ValueError("both load_in_8bit and load_in_4bit are set. "
-                         "only one of them should be set at a time")
-    
-    valid, input_steps = validate_steps(args.steps)
-    if valid is True:
-        if "fine-tuning" in input_steps:
+                            "only one of them should be set at a time")
+
+    if args.push_lm_responses_to_hf_hub is True:
+        if args.lm_response_dataset_id is None \
+            or args.hf_token is None:
+            raise ValueError("push_to_hub was set to True, but either or all of "
+                            "lm_response_dataset_id and hf_token are set to None")
+        else:
+            hf_hub = True
+
+    local_lm_responses = gen_local_lm_responses(
+        args.ft_model_id, args.load_in_8bit, args.load_in_4bit,
+        args.test_ds_id, args.test_ds_split, 
+        args.batch_infer_data_preprocess_bs, args.inference_bs, args.repeat,
+        args.ft_model_config_path, 
+    )
+
+    if hf_hub is True:
+        push_to_hf_hub(
```

Comment:
Is what format and data structure the results are being pushed?

Reply:
anything in Python dict, but the result could be found [here](https://huggingface.co/datasets/chansung/eval_dataset_test).
Also, I will add some comments for this